shared/chat/conversation-list/conversation-list-container.js

```diff
@@ -1,17 +1,189 @@
 // @flow
 import {namedConnect} from '../../util/container'
-import ConversationList from './conversation-list'
+import {memoize} from '../../util/memoize'
+import * as Types from '../../constants/types/chat2'
+import * as Constants from '../../constants/chat2'
+import * as Flow from '../../util/flow'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
+import {isMobile} from '../../constants/platform'
+import ConversationList, {type SmallTeamRowItem, type BigTeamChannelRowItem} from './conversation-list'
+import getFilteredRowsAndMetadata from '../inbox/container/filtered'
 
 type OwnProps = {|
-  onSelect?: () => void,
+  filter?: string,
+  onDone?: ?() => void,
+  onSelect: (conversationIDKey: Types.ConversationIDKey) => void,
+  onSetFilter?: (filter: string) => void,
+  selected: Types.ConversationIDKey, // mobile only
 |}
 
-const mapStateToProps = state => ({})
-const mapDispatchToProps = dispatch => ({})
-const mergeProps = (stateProps, dispatchProps, ownProps) => ({
-  rows: [],
+const notificationsTypeToNumber = (t: Types.NotificationsType): number => {
+  switch (t) {
+    case 'onAnyActivity':
+      return 1
+    case 'onWhenAtMentioned':
+      return 2
+    case 'never':
+      return 3
+    default:
+      Flow.ifFlowComplainsAboutThisFunctionYouHaventHandledAllCasesInASwitch(t)
+      return 0
+  }
+}
+
+const boolToNumber = (b: boolean): number => (b ? 1 : 0)
+
+const staleToNumber = (convTime: number, staleCutoff: number) => (convTime < staleCutoff ? 1 : 0)
+
+const getAWeekAgo = () => {
+  let t = new Date()
+  return t.setDate(t.getDate() - 7) // works fine for cross-boundary; returns a number
+}
+
+const getSortedConversationIDKeys = memoize(
+  (metaMap: Types.MetaMap): Array<{conversationIDKey: Types.ConversationIDKey, type: 'small' | 'big'}> => {
+    const staleCutoff = getAWeekAgo()
+    return metaMap
+      .valueSeq()
+      .toArray()
+      .sort((a, b) => {
+        // leveled order rules:
+        // 1. unmuted before muted
+        // 2. active conversations before inactive (has activity in the past week)
+        // 3. notification type: onAnyActivity before onWhenAtMentioned, before never
+        // 4. activity timestamp being the last tie breaker
+        const mutedBased = boolToNumber(a.isMuted) - boolToNumber(b.isMuted)
```

Comment:
if the goal is to see if they're both the same truthy vs falsey you can just be
```
const muteBased = !a.isMuted !== !b.isMuted
```

Reply:
I still need a number here to return though, so would still need to use boolToNumber.